projects/igniteui-angular/src/lib/grids/row.directive.ts

```diff
@@ -56,7 +57,12 @@ export class IgxRowDirective<T extends IgxGridBaseDirective & GridType> implemen
     @Input()
     public get rowData(): any {
         if (this.inEditMode) {
-            return merge({...this._rowData }, this.grid.transactions.getAggregatedValue(this.rowID, false));
+            return mergeWith(cloneValue(this._rowData), this.grid.transactions.getAggregatedValue(this.rowID, false),
```

Comment:
This could've used the `mergeObjects` from utils directly to skip one more array handler, but no biggie.